### PR TITLE
feat: allow adding custom headers with nuxt config

### LINKED ISSUE #77: allow adding custom headers with nuxt config
nuxt-community/axios-module#294

<div align="right"><sub><em>This question is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/http/issues/c15">#c15</a>)</em></sub></div>

### LINKED ISSUE #95: Docs example for setToken with request type
### Version

[v0.5.0](https://github.com/http/releases/tag/v0.5.0)

### Reproduction link

[https://http.nuxtjs.org/guide/advanced.html#settoken-token-type](https://http.nuxtjs.org/guide/advanced.html#settoken-token-type)

### Steps to reproduce

The example in the docs shows you can set it by request type but the functionality doesn't exist. Planned for future perhaps?

```
// Adds header: `Authorization: Bearer 123` to only post and delete requests
this.$http.setToken('123', 'Bearer', ['post', 'delete'])
```

### What is expected ?

Set token by request type

### What is actually happening?

Not possible

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/http/issues/c25">#c25</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
Resolve #77 
Resolve #95

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -111,23 +112,30 @@ export default (ctx, inject) => {
...
+      delete reqHeaders[h]
+    }
+    defaults.headers.common = { ...reqHeaders, ...defaults.headers.common }
+  }
+  <% } %>
 
   if (process.server) {
     // Don't accept brotli encoding because Node can't parse it
-    defaults.headers['accept-encoding'] = 'gzip, deflate'
+    defaults.headers.common['accept-encoding'] = 'gzip, deflate'
```
Comment (pi0) [inline lib/plugin.js:138]: Ky headers are flat: https://www.npmjs.com/package/ky#kyextenddefaultoptions :)

File: lib/module.js [hunk 2]
```diff
@@ -34,8 +35,21 @@ function httpModule (_moduleOptions) {
   // Default prefix
   const prefix = process.env.API_PREFIX || moduleOptions.prefix || '/'
 
+  // Headers
+  const headers = {
+    common: {
+      Accept: 'application/json, text/plain, */*'
```
Comment (pi0) [inline lib/module.js:41]: It is automatically set for Accept header. Other defaults was due to a SSR shared state bug in axios

File: lib/module.js [hunk 3]
```diff
@@ -34,8 +35,21 @@ function httpModule (_moduleOptions) {
...
+    get: {},
+    head: {},
+    post: {},
+    put: {},
+    patch: {}
+  }
+
   // Apply defaults
-  const options = {
+  const options = defu(moduleOptions, {
```
Comment (pi0) [inline lib/module.js:52]: Maybe we can simply use destructure to drop defu dep?

File: lib/plugin.js (hunk 1 above)
Comment (ricardogobbosouza) [reply lib/plugin.js]: Sure. fixed

File: lib/plugin.js [hunk 4]
```diff
@@ -111,19 +110,26 @@ export default (ctx, inject) => {
       ? '<%= options.browserBaseURL %>'
       : (process.env._HTTP_BASE_URL_ || '<%= options.baseURL %>')
 
+  const headers = <%= JSON.stringify(options.headers, null, 4) %>
```
Comment (pi0) [inline lib/plugin.js:113]: [editorconfig](https://github.com/nuxt/http/blob/dev/.editorconfig#L5) is set to 2

Comment (ricardogobbosouza) [reply lib/plugin.js]: Fixed

**Pull request merged**